js/stdlib/ts/ALGO.ts

```diff
@@ -1199,6 +1199,34 @@ const getApplicationInfoM = async (idn:BigNumber): Promise<OrExn<AppInfo>> => {
   return 'exn' in res ? res : { val: res.val.application };
 };
 
+const getDeletedApplicationInfoM = async (idn: BigNumber): Promise<OrExn<AppInfo>> => {
+  const dhead = 'getDeletedApplicationInfoM'
+  const indexer = await getIndexer();
+  const query = indexer.searchForTransactions()
+                       .txType('appl')
+                       .applicationID(bigNumberToNumber(idn))
+                       .limit(1);
+  const queryRes = await doQueryM_(dhead, query);
+
+  if ('val' in queryRes) {
+    const txn = queryRes.val.transactions[0];
+    const appTxn = txn['application-transaction'];
+    debug(dhead, {txn});
+    return { val: {
+      'id': bigNumberToBigInt(idn),
```

Comment:
You should read this from the txn

Reply:
The id will be set to 0 if we find the right txn, as you mentioned, so I think it doesn't make sense to do that.